draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -752,54 +790,50 @@ normative:
     target: https://www.sagemath.org
     date: 2019
 
-
 --- abstract
 
 This document specifies a number of algorithms that may be used to encode or hash an
-arbitrary string to a point on an Elliptic Curve.
+arbitrary string to a point on an elliptic curve.
 
 --- middle
 
 # Introduction {#introduction}
 
-Many cryptographic protocols require a procedure which encodes arbitrary input,
-e.g., a password, to a point on an elliptic curve (EC). This procedure is known
+Many cryptographic protocols require a procedure that encodes an arbitrary input,
+e.g., a password, to a point on an elliptic curve. This procedure is known
 as hashing to an elliptic curve. Prominent examples of cryptosystems that
 hash to elliptic curves include Simple Password Exponential Key Exchange
 {{J96}}, Password Authenticated Key Exchange {{BMP00}}, Identity-Based
 Encryption {{BF01}} and Boneh-Lynn-Shacham signatures {{BLS01}}.
 
-Unfortunately for implementors, the precise encoding which is suitable for a
+Unfortunately for implementors, the precise hash function that is suitable for a
 given scheme is not necessarily included in the description of the protocol.
 Compounding this problem is the need to pick a suitable curve for the specific
 protocol.
 
 This document aims to bridge this gap by providing a thorough set of
-recommendations across a range of implementations and curve types. We provide
-implementation and performance details for each mechanism, along with references
-to the security rationale behind each recommendation and guidance for
-applications not yet covered.
-
-Each algorithm conforms to a common interface, i.e., it encodes a bitstring
-{0, 1}^\* to a point on an elliptic curve E. For each variant, we describe the requirements for
-E to make it work. Sample code for each variant is presented in the
-appendix.  Unless otherwise stated, all elliptic curve points are assumed to be
-represented as affine coordinates, i.e., the pair (x, y) denotes a point on an
-elliptic curve (see {{bg-curves}}).
+recommended algorithms for a range of curve types. We provide
+implementation and performance details for each algorithm, describe
+the security rationale behind each recommendation, and give guidance for
+elliptic curves that are not explicitly covered.
+
+Each algorithm conforms to a common interface: it takes as input an arbitrary-length bit string
+and produces as output a point on an elliptic curve.
+Sample code for each algorithm is presented in {{samplecode}}.
```

Comment:
I'd prefer it be near the top rather than hidden at the end of the section.

Reply:
For now I removed the forward ref.

We could bring it back, but it seems unnecessary to have the stereotypical "this document is organized as follows..." paragraph as long as the section titles are descriptive, since there's an autogenerated table of contents.